### PR TITLE
Update Pose track duplicate/copy methods.

### PR DESCRIPTION
**Pull request by makiam**
Take in accоunt that Pose Track subtracks may be added from outside. Ex. Camera's filters parameters tracks
Original issue from Sourceforge thread: https://sourceforge.net/p/aoi/discussion/47783/thread/a299917f8b/

### CODE REVIEW

File: ArtOfIllusion/src/artofillusion/animation/PoseTrack.java [hunk 1]
```diff
@@ -85,8 +85,10 @@ public Track duplicate(Object obj)
     t.smoothingMethod = smoothingMethod;
     t.tc = tc.duplicate((ObjectInfo) obj);
     t.relative = relative;
-    t.theWeight = (WeightTrack) theWeight.duplicate(t);
-    t.subtracks = new Track [] {theWeight};
+    t.theWeight = theWeight.duplicate(t);
+    t.subtracks = new Track[this.subtracks.length];
+    for (int i = 0; i < this.subtracks.length; i++)
+      t.subtracks[i] = this.subtracks[i].duplicate(t);
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/animation/PoseTrack.java:91]: Perfect place for an 'enhanced for' 

`for (Track: this.subtracks)`

File: ArtOfIllusion/src/artofillusion/animation/PoseTrack.java [hunk 2]
```diff
@@ -103,8 +105,10 @@ public void copy(Track tr)
     smoothingMethod = t.smoothingMethod;
     tc = t.tc.duplicate(info);
     relative = t.relative;
-    theWeight = (WeightTrack) t.theWeight.duplicate(this);
-    subtracks = new Track [] {theWeight};
+    theWeight = t.theWeight.duplicate(this);
+    this.subtracks = new Track[t.subtracks.length];
+    for (int i = 0; i < this.subtracks.length; i++)
+      this.subtracks[i] = t.subtracks[i].duplicate(this);
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/animation/PoseTrack.java:111]: Same.

File: ArtOfIllusion/src/artofillusion/animation/PoseTrack.java (hunk 1 above)
Comment (makiam) [reply ArtOfIllusion/src/artofillusion/animation/PoseTrack.java:91]: I cannot use enhanced for becase i still need  i-index variable to put duplucates to target array.

Comment (Sailsman63) [reply ArtOfIllusion/src/artofillusion/animation/PoseTrack.java:91]: That's an annoying hole in the enhanced loop...

**Pull request merged**